src/qibolab/transpilers/placer.py

```diff
@@ -0,0 +1,261 @@
+        GM = nx.algorithms.isomorphism.GraphMatcher(self.connectivity, H)
+        i = 0
+        H.add_edge(circuit_repr[i][0], circuit_repr[i][1])
+        while GM.subgraph_is_monomorphic() == True:
+            result = GM
+            i += 1
+            H.add_edge(circuit_repr[i][0], circuit_repr[i][1])
+            GM = nx.algorithms.isomorphism.GraphMatcher(self.connectivity, H)
+            if self.connectivity.number_of_edges() == H.number_of_edges() or i == len(circuit_repr) - 1:
+                keys = list(result.mapping.keys())
+                keys.sort()
+                return {i: result.mapping[i] for i in keys}
+        keys = list(result.mapping.keys())
+        keys.sort()
+        return {i: result.mapping[i] for i in keys}
+
+
+class Random(Placer):
+    """
+    Random initialization with greedy policy, let a maximum number of 2-qubit
+    gates can be applied without introducing any SWAP gate
+
+    Attributes:
+        connectivity (networkx.Graph): chip connectivity.
+        samples (int): number of initial random layouts tested.
+    """
+
+    def __init__(self, connectivity, samples=100):
+        """Args:
+        connectivity (networkx.graph): chip connectivity.
+        samples (int): number of initial random layouts tested.
+        """
+        self.connectivity = connectivity
+        self.samples = samples
+
+    def __call__(self, circuit):
+        """Find an initial layout of the given circuit using random greedy algorithm.
+
+        Args:
+            circuit (qibo.models.Circuit): Circuit to be transpiled.
+        """
+        circuit_repr = create_circuit_repr(circuit)
+        nodes = self.connectivity.number_of_nodes()
+        keys = list(self.connectivity.nodes())
+        final_mapping = dict(zip(keys, list(range(nodes))))
+        final_graph = nx.relabel_nodes(self.connectivity, final_mapping)
+        final_cost = self.cost(final_graph, circuit_repr)
+        for _ in range(self.samples):
+            mapping = dict(zip(keys, random.sample(range(nodes), nodes)))
+            graph = nx.relabel_nodes(self.connectivity, mapping)
+            cost = self.cost(graph, circuit_repr)
+            if cost == 0:
+                return mapping
+            if cost < final_cost:
+                final_graph = graph
+                final_mapping = mapping
+                final_cost = cost
+        return final_mapping
+
+    @staticmethod
+    def cost(graph, circuit_repr):
+        """
+        Compute the cost associated to an initial layout as the lengh of the reduced circuit.
+
+        Args:
+            graph (networkx.Graph): current hardware qubit mapping.
+            circuit_repr (list): circuit representation.
+
+        Returns:
+            (int): lengh of the reduced circuit.
+        """
+        new_circuit = circuit_repr.copy()
+        while new_circuit != [] and (new_circuit[0][0], new_circuit[0][1]) in graph.edges():
+            del new_circuit[0]
+        return len(new_circuit)
```

Comment:
```suggestion
        for allowed, gate in enumerate(circuit_repr):
            if (gate[0], gate[1]) not in graph.edges():
                break
        return len(new_circuit) - allowed
```
In this way, you can avoid *"expensive" copies* and *mutation*.

Moreover, if you drop the number element in the third position (as suggested above), you can also replace `(gate[0], gate[1])` with just `gate`.